fix: use response-data to update server members

Droid00000 opened the pull request:
## Summary

This PR makes the cached data for member objects be immediately updated upon updating a member, via the updated member object returned by the REST-API.

## Fixed
I noticed that when using `#inspect`, all of the voice state attributes would always be `nil`. This is because `#inspect` was referencing the attributes as if they were instance variables, when they're actually methods.

swarley commented:
This is a good change, however we are doing a lot of repeating for `update_data(JSON.parse(API::Server.update_member(...)))`, maybe we do a private update function to centralize this to a single call

Droid00000 commented:
> This is a good change, however we are doing a lot of repeating for `update_data(JSON.parse(API::Server.update_member(...)))`, maybe we do a private update function to centralize this to a single call

Alright, I went and added `update_member_data` and `update_current_member_data` respectively, and updated everything

swarley commented:
Much cleaner, good change

Droid00000 commented:
> Much cleaner, good change

In the spirit of the changes done here, I'm going to update https://github.com/shardlab/discordrb/pull/350 in a bit since this PR affects that one pretty heavily, lol
